### ISSUE TITLE
Id fields duplicates for domain inheritance 

### ISSUE THREAD

**Issue by demon101** [grails/grails-data-mapping#1216]
in case of inheritance all classes in the hierarchy has id fields. 
For AccessType.PROPERTY doesn't bring any bugs. All side effects corrected by method overriding.
But for GORM 6.1 AccessType.FIELD used by default. At the result, some strange bugs occur:

> Caused by: java.lang.NullPointerException: null
> 	at org.hibernate.type.descriptor.java.AbstractTypeDescriptor.extractHashCode(AbstractTypeDescriptor.java:65)
> 	at org.hibernate.type.AbstractStandardBasicType.getHashCode(AbstractStandardBasicType.java:185)
> 	at org.hibernate.type.AbstractStandardBasicType.getHashCode(AbstractStandardBasicType.java:189)
> 	at org.hibernate.type.EntityType.getHashCode(EntityType.java:355)
> 	at org.hibernate.type.ComponentType.getHashCode(ComponentType.java:241)
> 	at org.hibernate.engine.spi.EntityKey.generateHashCode(EntityKey.java:59)
> 	at org.hibernate.engine.spi.EntityKey.<init>(EntityKey.java:54)
> 	at org.hibernate.internal.AbstractSessionImpl.generateEntityKey(AbstractSessionImpl.java:338)
> 	at org.hibernate.engine.internal.StatefulPersistenceContext.getDatabaseSnapshot(StatefulPersistenceContext.java:308)
> 	at org.hibernate.engine.internal.ForeignKeys.isTransient(ForeignKeys.java:238)
> 	at org.hibernate.event.internal.AbstractSaveEventListener.getEntityState(AbstractSaveEventListener.java:499)
> 	at org.hibernate.event.internal.DefaultSaveOrUpdateEventListener.performSaveOrUpdate(DefaultSaveOrUpdateEventListener.java:83)
> 	at org.hibernate.event.internal.DefaultSaveOrUpdateEventListener.onSaveOrUpdate(DefaultSaveOrUpdateEventListener.java:73)
> 	at org.grails.orm.hibernate.support.ClosureEventTriggeringInterceptor.onSaveOrUpdate(ClosureEventTriggeringInterceptor.java:126)
> 	at org.hibernate.internal.SessionImpl.fireSaveOrUpdate(SessionImpl.java:651)
> 	at org.hibernate.internal.SessionImpl.saveOrUpdate(SessionImpl.java:643)
> 	at org.hibernate.internal.SessionImpl.saveOrUpdate(SessionImpl.java:638)
> 	at org.grails.orm.hibernate.AbstractHibernateGormInstanceApi$_performSave_closure3.doCall(AbstractHibernateGormInstanceApi.groovy:247)
> 	at sun.reflect.GeneratedMethodAccessor566.invoke(Unknown Source)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at org.springsource.loaded.ri.ReflectiveInterceptor.jlrMethodInvoke(ReflectiveInterceptor.java:1427)
> 	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:98)
> 	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
> 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1225)
> 	at groovy.lang.ExpandoMetaClass.invokeMethod(ExpandoMetaClass.java:1125)
> 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1034)
> 	at groovy.lang.Closure.call(Closure.java:418)
> 	at org.codehaus.groovy.runtime.ConvertedClosure.invokeCustom(ConvertedClosure.java:54)
> 	at org.codehaus.groovy.runtime.ConversionHandler.invoke(ConversionHandler.java:124)
> 	at com.sun.proxy.$Proxy138.doInHibernate(Unknown Source)

The problem is: 
id field of child filled during saving, but persister.getIdentifier( x ) looking on id field of parent.  

Childe id field looks excessive and shouldn't be added.

```
def o = new Child().save()
print o.id // some number
print ((Parent) o).id // null
```

 
Bug can be solved either by filling by same value parent.id or by removing of duplicate field of childe   

### Task List

- [X] Steps to reproduce provided
- [X] Stacktrace (if present) provided
- [ ] Example that reproduces the problem uploaded to Github
- [ ] Full description of the issue provided (see below)

### Steps to Reproduce

1. TODO
2. TODO
3. TODO

### Expected Behaviour

Tell us what should happen

### Actual Behaviour

Tell us what happens instead

### Environment Information

- **Operating System**: TODO
- **GORM Version:** 6.1.11
- **Grails Version (if using Grails):** 3.3.9
- **JDK Version:** TODO

### Example Application

- TODO: link to github repository with example that reproduces the issue

**Comment by demon101** [comment]
> should be handled here https://github.com/grails/grails-data-mapping/blob/master/grails-datastore-gorm/src/main/groovy/org/grails/compiler/gorm/GormEntityTransformation.groovy#L459
> but may be a bug in Grails
> see https://github.com/grails/grails-core/blob/master/grails-core/src/main/groovy/org/grails/compiler/injection/DefaultGrailsDomainClassInjector.java#L222
> it is in one of those two places

**Comment by demon101** [comment]
First of all, both classes method executed. DefaultGrailsDomainClassInjector first, GormEntityTransformation second.

Problem in line
`ClassNode parent = GrailsASTUtils.getFurthestUnresolvedParent(classNode);`

For child class it returns same class node, not a parent. At result, id will be inserted in child

**Comment by demon101** [comment]
The issue can be closed after https://github.com/grails/grails-core/pull/11254

**Comment by demon101** [comment]
Fix merged, will be part of Grails 3.3.10
